### DISCUSSION TITLE
Bug: koishi启动时出现错误(Debian 11)

### DISCUSSION THREAD

**Discussion by nrxboat** [koishijs/koishi#944] [Q&A] [answered]
### Describe the bug

使用npm start命令启动（安装koishi后的首次启动）时出现报错现象，且访问对应端口未能显示控制台

### Steps to reproduce

Debian11下按照“创建模板项目”中的“使用包管理器创建”部署koishi，然后使用npm start启动

另外在replit这个平台上部署时也存在这个问题

### Expected behavior

控制台正常显示

### Screenshots

以下为安装koishi时出现的异常日志：
npm WARN deprecated sourcemap-codec@1.4.8: Please use @jridgewell/sourcemap-codec instead

以下为koishi启动时的部分异常日志：
2023-01-20 05:44:27 [W] assets missing configuration "selfUrl", fallback to "file:" scheme2023-01-20 05:44:27 [I] app server listening at http://localhost:5140
2023-01-20 05:44:33 [W] market timeout of 5000ms exceeded
2023-01-20 05:44:33 [W] market timeout of 5000ms exceeded
2023-01-20 05:44:58 [W] market AxiosError: timeout of 30000ms exceeded
                            at RedirectableRequest.handleRequestTimeout (/root/koishi-bot/node_modules/axios/dist/node/axios.cjs:2504:16)
                            at RedirectableRequest.emit (node:events:513:28)
                            at Timeout.<anonymous> (/root/koishi-bot/node_modules/follow-redirects/index.js:169:12)
                            at listOnTimeout (node:internal/timers:564:17)
                            at process.processTimers (node:internal/timers:507:7)

### Versions

- OS: Debian 11 
- Platform: 尚未部署 
- Node version: 18.13.0LTS 
- Koishi version:  5.2.7

**Comment by ilharp** [comment]
此错误是网络质量原因所致，你的网络环境无法连接到 Koishi 的插件市场。Koishi 插件市场之外的其他功能仍然可用。

**Comment by nrxboat** [comment]
> 此错误是网络质量原因所致，你的网络环境无法连接到 Koishi 的插件市场。Koishi 插件市场之外的其他功能仍然可用。

vps区域和replit均位于国外，推测应该不是这个原因
另外可能是我的描述有点问题，我再改改描述

**Comment by ilharp** [comment]
Koishi 插件市场默认使用大陆镜像源，国外无法连接是预期的。你可以修改 `.env` 文件和 market 插件的配置以选择离你所在地区最近的镜像。

**Comment by nrxboat** [comment]
尝试修改.env后发现问题仍然存在，问题应该不在于网络问题
log中虽显示了成功监听端口，但实际上无法打开控制台
在已提交的日志中
2023-01-20 05:44:27 [W] assets missing configuration "selfUrl", fallback to "file:"
这条日志可能能够反映出现问题的原因


网上搜了一下，现有的搭建教程几乎全是在windows上部署的koishi，剩下的也基本是安卓用termux建chroot容器安装的，在linux上安装koishi的例子几乎没见过

**Comment by nrxboat** [comment]
再次尝试启动koishi，结果如下：
![Screenshot_2023-01-20-23-45-23-66](https://user-images.githubusercontent.com/69232821/213742884-4f02a801-5dd5-4872-842e-83a7ca10a62a.png)

按ctrl+c后，结果如下：
![Screenshot_2023-01-20-23-45-34-20](https://user-images.githubusercontent.com/69232821/213743017-d7c1a86a-37ca-4895-ad0d-d87f53dca888.png)

**Comment by ilharp** [comment, marked as answer]
> 问题应该不在于网络问题

没看到「无法显示控制台」，是我的疏忽。控制台内无法显示插件市场是网络问题，无法显示控制台是因为你未按照官方文档中 [服务器部署](https://koishi.chat/manual/recipe/server.html) 的步骤操作所致。

> 这条日志可能能够反映出现问题的原因

这条日志是意图的行为，并非问题的原因。

> 网上搜了一下，现有的搭建教程几乎全是在windows上部署的koishi，剩下的也基本是安卓用termux建chroot容器安装的，在linux上安装koishi的例子几乎没见过

这是因为在 Linux 上使用 Koishi 的大多数用户遇到问题都可以自己解决，导致了互联网上在 Linux 上运行 Koishi 的资源较少的现象。Koishi 团队绝大多数成员的 Koishi 生产环境均为 Linux，Linux 在目前仍是长期稳定运行 Koishi 的最佳选择。

---

鉴于本 issue 的讨论内容并非 Koishi 的 bug，本 issue 将被锁定后移至 discussion。后续可前往对应的 discussion 继续交流，如果有其他反馈内容请创建新的 issue。

**Reply by shigma** [reply]
现在 Koishi 的官方机器人就在 Linux 上运行。

**Reply by nrxboat** [reply]
感谢回答，问题成功解决，是我没有仔细查看文档，很抱歉打扰了您

**Comment by Mcayear** [comment]
我也遇到了这个问题并且毫无头绪
![image](https://github.com/koishijs/koishi/assets/48203962/a972eafc-0300-4586-ae0a-2d01c73028cf)


```
^[[38;5;8m2023-05-17 22:22:47^[[0m [I] ^[[38;5;78;1mapp^[[0m apply plugin ^[[38;5;78mconfig^[[0m       
^[[38;5;8m2023-05-17 22:22:47^[[0m [I] ^[[38;5;78;1mapp^[[0m apply plugin ^[[38;5;78mconsole^[[0m      
^[[38;5;8m2023-05-17 22:22:47^[[0m [I] ^[[38;5;78;1mapp^[[0m apply plugin ^[[38;5;78mdataview^[[0m     
^[[38;5;8m2023-05-17 22:22:47^[[0m [I] ^[[38;5;78;1mapp^[[0m apply plugin ^[[38;5;78mexplorer^[[0m 
^[[38;5;8m2023-05-17 22:22:47^[[0m [I] ^[[38;5;78;1mapp^[[0m apply plugin ^[[38;5;78mlogger^[[0m  
^[[38;5;8m2023-05-17 22:22:47^[[0m [I] ^[[38;5;78;1mapp^[[0m apply plugin ^[[38;5;78minsight^[[0m 
^[[38;5;8m2023-05-17 22:22:47^[[0m [I] ^[[38;5;78;1mapp^[[0m apply plugin ^[[38;5;78mmarket^[[0m  
^[[38;5;8m2023-05-17 22:22:47^[[0m [I] ^[[38;5;78;1mapp^[[0m apply plugin ^[[38;5;78msandbox^[[0m 
^[[38;5;8m2023-05-17 22:22:47^[[0m [I] ^[[38;5;78;1mapp^[[0m apply plugin ^[[38;5;78mstatus^[[0m 
^[[38;5;8m2023-05-17 22:22:47^[[0m [I] ^[[38;5;78;1mapp^[[0m apply plugin ^[[38;5;78mgroup:storage^[[0m
^[[38;5;8m2023-05-17 22:22:47^[[0m [I] ^[[38;5;78;1mapp^[[0m apply plugin ^[[38;5;78mdatabase-sqlite^[[0m
^[[38;5;8m2023-05-17 22:22:47^[[0m [I] ^[[38;5;78;1mapp^[[0m apply plugin ^[[38;5;78massets-local^[[0m   
^[[38;5;8m2023-05-17 22:22:47^[[0m [I] ^[[38;5;78;1mapp^[[0m apply plugin ^[[38;5;78mgroup:adapter^[[0m  
^[[38;5;8m2023-05-17 22:22:47^[[0m [I] ^[[38;5;78;1mapp^[[0m apply plugin ^[[38;5;78mgocqhttp^[[0m       
^[[38;5;8m2023-05-17 22:22:48^[[0m [I] ^[[38;5;78;1mapp^[[0m server listening at ^[[38;5;78mhttp://0.0.0.0:5140^[[0m
^[[38;5;8m2023-05-17 22:23:18^[[0m [W] ^[[38;5;178;1mmarket^[[0m AxiosError: timeout of 30000ms exceeded            
                            at RedirectableRequest.handleRequestTimeout (/koishi/node_modules/axios/dist/node/axios.cjs:2504:16)
                            at RedirectableRequest.emit (node:events:513:28)                                                    
                            at Timeout.<anonymous> (/koishi/node_modules/follow-redirects/index.js:169:12)                      
                            at listOnTimeout (node:internal/timers:569:17)                                                      
                            at process.processTimers (node:internal/timers:512:7)     
```

测试与npm中心的连接也是通的
![image](https://github.com/koishijs/koishi/assets/48203962/6814e9e6-8078-4158-9ac0-91b63fd62ee5)

**Reply by ilharp** [reply]
该问题是由于网络环境不佳所致，更换更优质的网络环境即可解决。

**Reply by ilharp** [reply]
默认情况下（新安装情况下），插件市场并不使用 npm 作为 registry，测试 npm 的网络连接性并不能证明插件市场的可访问性。

**Reply by Mcayear** [reply]
> 该问题是由于网络环境不佳所致，更换更优质的网络环境即可解决。

我用 `wget https://registry.npmjs.org/` 是可以正常下载的。

这是`vi index.html`查看下载的内容
![image](https://github.com/koishijs/koishi/assets/48203962/d7eb7609-3591-49f5-ae1a-cdbb587f9841)

**Reply by Mcayear** [reply]
> 默认情况下（新安装情况下），插件市场并不使用 npm 作为 registry，测试 npm 的网络连接性并不能证明插件市场的可访问性。

现行有清晰点的解决方案吗？第一次使用koishi不是很清楚。

比如，更换npm源是正确的吗？

**Reply by ilharp** [reply]
如果你的接入点位于大陆，那么你需要更换至更优质的网络环境。Koishi 在默认情况下针对大陆的网络环境进行优化，如不能访问则说明你的网络环境质量不佳。

如果你的接入点位于大陆之外，可以尝试 [这个帖子](https://forum.koishi.xyz/t/topic/833/2) 中的方法。

**Reply by Mcayear** [reply]
> 如果你的网络环境位于大陆，那么你需要更换至更优的网络环境。
> 
> 如果你的接入点位于大陆之外，可以尝试 [这个帖子](https://forum.koishi.xyz/t/topic/833/2) 中的方法。

我测试出来了，是因为无法连接 `https://registry.koishi.chat/index.json`
![image](https://github.com/koishijs/koishi/assets/48203962/1ddb757e-08cd-4ad8-9199-a5f3904f89e0)


请问这个索引有其它的镜像链接吗？万般感谢！

**Reply by ilharp** [reply]
我们正在调查插件市场源可访问性下降的相关问题。这可能需要几分钟，请稍作等待。

**Reply by Mcayear** [reply]
> 我们正在调查插件市场源可访问性下降的相关问题。这可能需要几分钟，请稍作等待。

好的，谢谢。我刚刚咨询了十堰的服务商，他说现在地区性墙很普遍。目前我替换成自己的临时json已经可以加载插件市场了。
![image](https://github.com/koishijs/koishi/assets/48203962/5a757338-9d7b-4bff-a3b8-dde9bc79ca0e)

**Reply by ilharp** [reply]
我们已对插件市场源可访问性下降的相关问题做出了对应，相关更改可能需要几小时才会生效。如果更改在几小时后仍未生效，则可使用搜索引擎搜索「清除 DNS 缓存」相关的内容并进行尝试。
